server/jamovi/tests/utils.py

```diff
@@ -0,0 +1,131 @@
+from typing import Sequence
+from typing import TypeAlias
+from typing import Iterable
+
+from importlib import resources
+
+import csv
+import math
+from uuid import uuid4
+import itertools
+
+import pytest
+
+from jamovi.server.dataset.dataset import DataSet
+from jamovi.server.dataset.duckdataset import DuckDataSet
+from jamovi.server.dataset.duckcolumn import DuckColumn
+from jamovi.server.dataset.duckcolumn import DuckLevel
+from jamovi.server.dataset.duckcolumn import Level
+from jamovi.server.dataset import DataType
+from jamovi.server.dataset import MeasureType
+
+
+CellValue: TypeAlias = str | int | float
+
+NAN = float("nan")
+NAN_INT = -2147483648
+
+
+def equals(x: CellValue, y: CellValue) -> bool:
+    """test if two cell values are equal"""
+    if isinstance(x, float) and isinstance(y, float):
+        if math.isnan(x):
+            return math.isnan(y)
+        if math.isnan(y):
+            return False
+        return pytest.approx(x) == y
+    else:
+        return x == y
+
+
+def load(dataset: DataSet, reader: Iterable[Iterable[str]]) -> None:
+    """load data into a dataset"""
+    skip_1st = itertools.islice(reader, 1, None)
+    for row_no, row in enumerate(skip_1st):
+        if row_no >= dataset.row_count:
+            dataset.set_row_count(row_no + 1)
+        for col_no, value in enumerate(row):
+            column = dataset[col_no]
+            if column.data_type is DataType.DECIMAL:
+                value = float(value)
+            elif column.data_type is DataType.INTEGER:
+                value = int(value)
+            elif column.has_levels and column.data_type is DataType.TEXT:
+                value = column.get_value_for_label(value)
+            column.set_value(row_no, value, True)
+
+
+def level_must_equal(level: DuckLevel, values: dict):
+    """tests whether a level has certain values"""
+    for name, value in values.items():
+        assert getattr(level, name) == value
```

Comment:
Best to leave the assertion to the test (or else, at least name the function in a way that it's clear that it's asserting)